packages/react-components/react-toast/src/state/vanilla/toaster.ts

```diff
@@ -73,9 +80,17 @@ export class Toaster {
       return;
     }
 
-    this.listeners.set(eventType, callback);
+    const listener: ToastListenerMap[TType] = (e: CustomEvent<CommonToastDetail>) => {
+      if (e.detail.toasterId !== this.toasterId) {
+        return;
+      }
+
+      callback(e as CustomEvent<ShowToastEventDetail>);
```

Comment:
open to suggestions to remove this casting

Reply:
This is a tough one actually. It goes back to the problem of `unions` and `intersections` fighting over arguments. I'd suggest as a first step to instead of declaring a map of event listeners like `ToastListenerMap` create a event map like `ToastEventMap` to avoid passing those generics through the event listener function declaration. Not sure if it'll fix this 100% though